apis/management.cattle.io/v3/authn_types.go

```diff
@@ -111,11 +111,36 @@ type GithubConfigTestOutput struct {
 }
 
 type GithubConfigApplyInput struct {
-	GithubConfig GithubConfig `json:"githubConfig, omitempty"`
+	GithubConfig GithubConfig `json:"githubConfig,omitempty"`
 	Code         string       `json:"code,omitempty"`
 	Enabled      bool         `json:"enabled,omitempty"`
 }
 
+type AzureADConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	AuthConfig        `json:",inline" mapstructure:",squash"`
+
+	AzureADEndpoint      string `json:"azureADEndpoint,omitempty" norman:"default=https://login.microsoftonline.com/,required,notnullable"`
+	AzureADGraphEndpoint string `json:"azureADGraphEndpoint,omitempty" norman:"required,notnullable"`
+	AzureADTokenEndpoint string `json:"azureADTokenEndpoint,omitempty" norman:"required,notnullable"`
+	AzureADAuthEndpoint  string `json:"azureADAuthEndpoint,omitempty" norman:"required,notnullable"`
+	TenantID             string `json:"tenantID,omitempty" norman:"required,notnullable"`
+	ApplicationID        string `json:"applicationID,omitempty" norman:"required,notnullable"`
+	ApplicationSecret    string `json:"applicationSecret,omitempty" norman:"required,notnullable,type=password"`
+	ServerURL            string `json:"serverURL,omitempty" norman:"required,notnullable"`
```

Comment:
SInce this field is the one exception (it is indicating rancher's url), we can add Rancher to its name. I'm cool with either RancherServerURL or rancherURL.

Reply:
in the rancher API, the convention for acronyms is to camelcase the acronym, not capitalize all of it.
so, should be like: `json:"serverUrl,omitempty"`